modes/basic-dev-mode/src/index.js

```diff
@@ -135,8 +135,11 @@ function modeFactory({ modeConfiguration }) {
       ]);
     },
     onModeExit: ({ servicesManager }) => {
```

Comment:
I'd suggest just making the toolGroupService be auto-destroy on mode exit - that prevents needing to know about it in the onModeExit, and the entire method can then be dropped.